docs/architecture/adr-040-storage-and-smt-state-commitments.md

```diff
@@ -110,6 +111,52 @@ We need to be able to process transactions and roll-back state updates if a tran
 
 We identified use-cases, where modules will need to save an object commitment without storing an object itself. Sometimes clients are receiving complex objects, and they have no way to prove a correctness of that object without knowing the storage layout. For those use cases it would be easier to commit to the object without storing it directly.
 
+
+### Remove MultiStore
+
+IAVL based store adds additional layer in the SDK store construction - the `MultiStore` structure. The multi store exists to support the modularity of the Cosmos SDK - each module is using it's own instance of IAVL, but in the current implementation, all instances share the same database.
+The latter indicates, however, that the implementation doesn't provide true modularity. Instead it causes problems related to race condition and sync problems (eg: [\#6370](https://github.com/cosmos/cosmos-sdk/issues/6370)).
+
+We propose to remove the MultiStore concept from the SDK, and use a single instance of `SC` and `SS` in a `rootStore` object. To avoid confusions, we should rename `MultiStore` interface to `RootStore` interface and make sure that the `rootStore` object will implement it.
```

Comment:
```suggestion
We propose to remove the multistore concept from the SDK, and to use a single instance of `SC` and `SS` in a `rootStore` object. To avoid confusion, we should rename the `MultiStore` interface to `RootStore` and make sure that the `rootStore` object implements `RootStore`.
```